yang/vyatta-protocols-frr-link-params-v1.yang

```diff
@@ -0,0 +1,229 @@
+					bandwidth is defined to be residual bandwidth minus the measured bandwidth used for the actual
+					forwarding of non-RSVP-TE LSP packets.  For a bundled link, available bandwidth is defined to be
+					the sum of the component link available bandwidths minus the measured bandwidth used for the actual
+					forwarding of non-RSVP-TE Label Switched Paths packets [draft-ietf-isis-te-metrics-extension-03].";
+				must "../max-bandwidth" {
+					error-message "Maximum Bandwidth must be set";
+				}
+				must "../available-bandwidth <= ../max-bandwidth" {
+					error-message "Available Bandwidth cannot be greater than Maximum Bandwidth";
+				}
+			}
+			leaf delay {
+				type uint32 {
+					range "0..16777215";
+				}
+				units "micro-second";
+				configd:help "Unidirectional Average Link Delay (micro-second)";
+				description "Unidirectional Average Link Delay";
+			}
+			leaf delay-variation {
+				type uint32 {
+					range "0..16777215";
+				}
+				units "micro-second";
+				configd:help "Unidirectional Link Delay Variation (micro-second)";
+				description "Unidirectional Link Delay Variation";
+			}
+			leaf max-bandwidth {
+				type uint64;
+				units "Bytes/second";
+				configd:help "Maximum bandwidth that can be used (Bytes/second)";
+				description "Maximum bandwidth that can be used";
+			}
+			leaf max-reserved-bandwidth {
+				type uint64;
+				units "Bytes/second";
+				configd:help "Maximum bandwidth that may be reserved (Bytes/second)";
+				description "Maximum bandwidth that may be reserved. It is the bandwidth that can
+					theoretically be reserved";
+				must "../max-bandwidth" {
+					error-message "Maximum Bandwidth must be set";
+				}
+				must "../max-reserved-bandwidth <= ../max-bandwidth" {
+					error-message "Maximum Reservable Bandwidth cannot be greater than Maximum Bandwidth";
+				}
+			}
+			leaf mpls-te-metric {
+				type uint32 {
+					range "1..max";
+				}
+				configd:help "Link metric for MPLS-TE purpose";
+				description "Link metric for MPLS-TE purpose";
+			}
+			leaf neighbor {
+				type types:ipv4-address;
+				configd:help "Configure remote ASBR information (Neighbor IP address and AS number)";
+				description "Configure remote ASBR information (Neighbor IP address and AS number)";
+			}
+			leaf packet-loss {
+				type uint32 {
+					range 0..100;
+				}
+				units "Percentage";
+				configd:help "Unidirectional Link Packet Loss (Percentage)";
+				description "Unidirectional Link Packet Loss (Percentage)";
+			}
+			leaf residual-bandwidth {
+				type uint64;
+				units "Bytes/second";
+				configd:help "Unidirectional Residual Bandwidth (Bytes/second)";
+				description "Unidirectional Residual Bandwidth. For a link or forwarding adjacency, residual
+					bandwidth is defined to be Maximum Bandwidth [RFC3630] minus the bandwidth currently allocated
+					to RSVP-TE LSPs. For a bundled link, residual bandwidth is defined to be the sum of the
```

Comment:
Where is the bandwidth allocated to RSVP-TE LSPs defined? What if there is a mismatch?

Reply:
You aren't convincing me that you fully understand what all these fields mean, nor how they relate to each other. I am not convinced that this data model is complete / consistent, and until I am persuaded that it is, I cannot approve.